### ISSUE TITLE
Incorrectly inferred Pandas indexing return type

### ISSUE THREAD

**Issue by kormalev** [astral-sh/ty#851]
### Summary

ty incorrectly thinks Pandas boolean indexing returns Series.

Reproducible example:
```python
import pandas as pd


def filter_data(data: pd.DataFrame) -> pd.DataFrame:
    return data[(data["a"] == 1)]


data = pd.DataFrame(
    {
        "a": [1, 2, 3, 4],
        "b": [4, 3, 2, 1],
    }
)

print(filter_data(data))
```

```text
> ty version
ty 0.0.1-alpha.14 (3ececb07e 2025-07-08)
> ty check pandas-demo.py
WARN ty is pre-release software and not ready for production use. Expect to encounter bugs, missing features, and fatal errors.
error[invalid-return-type]: Return type does not match returned value
 --> pandas-demo.py:4:40
  |
4 | def filter_data(data: pd.DataFrame) -> pd.DataFrame:
  |                                        ------------ Expected `DataFrame` because of return type
5 |     return data[data["a"] == 1]
  |            ^^^^^^^^^^^^^^^^^^^^ expected `DataFrame`, found `Series[Any]`
  |
info: rule `invalid-return-type` is enabled by default

Found 1 diagnostic
```

Pandas version is 2.3.0

**Comment by carljm** [comment]
I think this is because `Scalar` is a type alias defined using `TypeAlias`, which we don't support yet (#544), which means we match the wrong overload of `__getitem__`

**Comment by carljm** [comment]
Confirmed this will be fixed by https://github.com/astral-sh/ruff/pull/21394
